pkg/component/certmanagement/controller.go

```diff
@@ -0,0 +1,333 @@
+					Verbs:     []string{"get", "list", "update", "watch"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"secrets"},
+					Verbs:     []string{"get", "list", "update", "watch", "create", "delete"},
+				},
+				{
+					APIGroups: []string{"cert.gardener.cloud"},
+					Resources: []string{
+						"issuers", "issuers/status",
+						"certificates", "certificates/status",
+						"certificaterevocations", "certificaterevocations/status",
+					},
+					Verbs: []string{"get", "list", "update", "watch", "create", "delete"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"events"},
+					Verbs:     []string{"create", "patch"},
+				},
+				{
+					APIGroups: []string{"apiextensions.k8s.io"},
+					Resources: []string{"customresourcedefinitions"},
+					Verbs:     []string{"get", "list", "update", "create", "watch"},
+				},
+			},
+		}
+		clusterRoleBinding = &rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: clusterRole.Name,
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "ClusterRole",
+				Name:     clusterRole.Name,
+			},
+			Subjects: []rbacv1.Subject{{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      serviceAccount.Name,
+				Namespace: serviceAccount.Namespace,
+			}},
+		}
+		role = &rbacv1.Role{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resourceName,
+				Namespace: d.namespace,
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups: []string{"coordination.k8s.io"},
+					Resources: []string{"leases"},
+					Verbs:     []string{"create"},
+				},
+				{
+					APIGroups:     []string{"coordination.k8s.io"},
+					Resources:     []string{"leases"},
+					ResourceNames: []string{"cert-controller-manager-controllers"},
+					Verbs:         []string{"get", "watch", "update"},
+				},
+				{
+					APIGroups: []string{"dns.gardener.cloud"},
+					Resources: []string{"dnsentries"},
+					Verbs:     []string{"get", "list", "update", "watch", "create", "delete"},
+				},
+			},
+		}
+		roleBinding = &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      role.Name,
+				Namespace: d.namespace,
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      rbacv1.ServiceAccountKind,
+					Name:      serviceAccount.Name,
+					Namespace: serviceAccount.Namespace,
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "Role",
+				Name:     role.Name,
+			},
+		}
+
+		deployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resourceName,
+				Namespace: d.namespace,
+				Labels: gardenerutils.MergeStringMaps(getDeploymentLabels(), map[string]string{
+					resourcesv1alpha1.HighAvailabilityConfigType: resourcesv1alpha1.HighAvailabilityConfigTypeController,
+				}),
+			},
+			Spec: appsv1.DeploymentSpec{
+				Replicas:             ptr.To[int32](1),
+				RevisionHistoryLimit: ptr.To[int32](5),
```

Comment:
```suggestion
				RevisionHistoryLimit: ptr.To[int32](2),
```

[ref](https://github.com/gardener/gardener/blob/master/docs/development/component-checklist.md#observability--operations-productivity)